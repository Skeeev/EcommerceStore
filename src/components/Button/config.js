export const ButtonType = {
  Solid: 'solid',
  Clear: 'clear',
  Outline: 'outline'
};
