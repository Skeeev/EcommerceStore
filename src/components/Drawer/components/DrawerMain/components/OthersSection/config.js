export const shareOptions = {
  url: 'https://github.com/Skeeev/EcommerceStore',
  message: 'Repository with source code',
  failOnCancel: false
};
