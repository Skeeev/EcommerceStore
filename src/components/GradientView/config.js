export const gradientColors = [
  'rgb(185, 216, 230)',
  'rgb(196, 180, 208)',
  'rgb(241, 195, 180)',
  'rgb(242, 213, 184)'
];

export const gradientLocations = [0, 0.4, 0.8, 1];
