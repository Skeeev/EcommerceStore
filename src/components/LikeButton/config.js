import { LayoutAnimation } from 'react-native';

export const heartAnimation = LayoutAnimation.create(
  200,
  LayoutAnimation.Types.linear,
  LayoutAnimation.Properties.scaleXY
);
