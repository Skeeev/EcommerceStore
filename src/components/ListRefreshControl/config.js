import { RefreshControl } from 'react-native';

import { colors } from 'src/assets/styles/colors';

export const appearanceProps = {
  colors: [colors.bostonBlue],
  size: RefreshControl?.SIZE?.LARGE
};
