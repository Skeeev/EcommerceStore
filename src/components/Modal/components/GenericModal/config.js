import { colors } from 'src/assets/styles/colors';

export const icons = {
  success: {
    name: 'check-circle',
    color: colors.feijoa
  },
  warning: {
    name: 'exclamation-circle',
    color: colors.manhattan
  },
  error: {
    name: 'times-circle',
    color: colors.froly
  }
};
