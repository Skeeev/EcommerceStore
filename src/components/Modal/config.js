import { LoginModal, InternetConnectionErrorModal } from './components';

export const ModalName = {
  Login: 'login',
  InternetConnectionError: 'internetConnectionError'
};

export const ModalType = {
  Success: 'success',
  Warning: 'warning',
  Error: 'error'
};

export const modals = {
  [ModalName.Login]: LoginModal,
  [ModalName.InternetConnectionError]: InternetConnectionErrorModal
};
