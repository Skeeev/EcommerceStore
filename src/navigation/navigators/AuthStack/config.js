import { ScreenName } from 'src/constants/navigationScreenNames';
import { LoginScreen, SignUpScreen } from 'src/screens';
import { MainDrawer } from 'src/navigation/navigators';

export const navigatorScreenConfigs = [
  {
    name: ScreenName.Login,
    component: LoginScreen
  },
  {
    name: ScreenName.SignUp,
    component: SignUpScreen
  },
  {
    name: ScreenName.MainDrawer,
    component: MainDrawer
  }
];
