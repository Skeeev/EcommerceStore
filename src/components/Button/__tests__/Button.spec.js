import React from 'react';
import { shallow } from 'enzyme';

import { getComponentDisplayName } from 'src/utils/displayName';
import { ButtonType } from '../config';
import { Button } from '..';

describe(`${getComponentDisplayName(Button)} component`, () => {
  const renderComponent = props => shallow(<Button {...props} />);

  describe('renders', () => {
    describe('correctly for type', () => {
      it(ButtonType.Solid, () => {
        const component = renderComponent({
          type: ButtonType.Solid
        });

        expect(component).toMatchSnapshot();
      });

      it(ButtonType.Clear, () => {
        const component = renderComponent({
          type: ButtonType.Clear
        });

        expect(component).toMatchSnapshot();
      });

      it(ButtonType.Outline, () => {
        const component = renderComponent({
          type: ButtonType.Outline
        });

        expect(component).toMatchSnapshot();
      });
    });

    it('null for unknown type', () => {
      const component = renderComponent({
        type: 'dummy'
      });

      expect(component.isEmptyRender()).toBeTruthy();
    });
  });
});
