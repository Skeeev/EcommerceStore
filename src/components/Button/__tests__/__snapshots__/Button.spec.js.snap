// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component renders correctly for type clear 1`] = `
<TouchableOpacity
  accessibilityRole="button"
  activeOpacity={0.5}
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "justifyContent": "center",
      "padding": 10,
    }
  }
/>
`;

exports[`Button component renders correctly for type outline 1`] = `
<TouchableOpacity
  accessibilityRole="button"
  activeOpacity={0.5}
  style={
    Object {
      "alignItems": "center",
      "borderColor": "#318cbf",
      "borderStyle": "solid",
      "borderWidth": 1,
      "flexDirection": "row",
      "justifyContent": "center",
      "padding": 10,
    }
  }
/>
`;

exports[`Button component renders correctly for type solid 1`] = `
<TouchableOpacity
  accessibilityRole="button"
  activeOpacity={0.5}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#318cbf",
      "elevation": 3,
      "flexDirection": "row",
      "justifyContent": "center",
      "padding": 10,
      "shadowColor": "#000000",
      "shadowOffset": Object {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.25,
      "shadowRadius": 3.84,
    }
  }
/>
`;
